### PR TITLE
zh-cn: update the translation of Set

### LINKED ISSUE #15591: the common " // 大小写敏感 & duplicate ommision  " the 【 ommision  】 should be omission
### MDN URL

https://developer.mozilla.org/zh-CN/docs/Web/JavaScript/Reference/Global_Objects/Set

### 与此议题相关的小节或标题？

_No response_

### 哪些信息是错误、无用或不完善的？

 the common " // 大小写敏感 & duplicate ommision  " the 【 ommision  】 should be omission

### 你期望看到什么？

 the common " // 大小写敏感 & duplicate ommision  " the 【 ommision  】 should be omission

### 你有什么支持的链接、参考或引用？

_No response_

### 你还想补充什么内容？

_No response_

### MDN metadata

<details>
<summary>Page report details</summary>

* Folder: `zh-cn/web/javascript/reference/global_objects/set`
* MDN URL: https://developer.mozilla.org/zh-CN/docs/Web/JavaScript/Reference/Global_Objects/Set
* GitHub URL: https://github.com/mdn/translated-content/blob/main/files/zh-cn/web/javascript/reference/global_objects/set/index.md
* Last commit: https://github.com/mdn/translated-content/commit/665ae152f17af91a88860160ceb931323b320408
* Document last modified: 2023-07-29T10:39:27.000Z

</details>

### PR DESCRIPTION
**Pull request by yin1999**
### Description

update the translation of Set

### Related issues and pull requests

Fixes: #15591

### CODE REVIEW

File: files/zh-cn/web/javascript/reference/global_objects/set/index.md [hunk 1]
```diff
@@ -3,131 +3,181 @@ title: Set
...
+{{JSRef}}
 
-**`Set`** 对象允许你存储任何类型的唯一值，无论是{{ Glossary('Primitive', '原始值') }}或者是对象引用。
+**`Set`** 对象允许你存储任何类型（无论是{{Glossary("Primitive", "原始值")}}还是对象引用）的唯一值。
 
-## 简述
+## 描述
 
-`Set`对象是值的集合，你可以按照插入的顺序迭代它的元素。Set 中的元素只会**出现一次**，即 Set 中的元素是唯一的。
+`Set` 对象是值的合集（collection）。集合（set）中的元素只会**出现一次**，即集合中的元素是唯一的。你可以按照插入顺序迭代集合中的元素。*插入顺序*对应于 [`add()`](/zh-CN/docs/Web/JavaScript/Reference/Global_Objects/Set/add) 方法成功（即，调用 `add()` 方法时集合中不存在相同的元素）将每一个元素插入到集合中的顺序。
```
Comment (yin1999) [inline files/zh-cn/web/javascript/reference/global_objects/set/index.md:12]:
```suggestion
`Set` 对象是值的合集（collection）。集合（set）中的元素**只会出现一次**，即集合中的元素是唯一的。你可以按照插入顺序迭代集合中的元素。*插入顺序*对应于 [`add()`](/zh-CN/docs/Web/JavaScript/Reference/Global_Objects/Set/add) 方法成功将每一个元素插入到集合中（即，调用 `add()` 方法时集合中不存在相同的元素）的顺序。
```

File: files/zh-cn/web/javascript/reference/global_objects/set/index.md [hunk 2]
```diff
@@ -3,131 +3,181 @@ title: Set
...
+
+规范要求集合的实现是“对合集中的元素的访问时间是与集合中元素的数量呈次线性关系的”。因此，它可以在内部表示为哈希表（查找的时间复杂度为 O(1)）、搜索树（查找的时间复杂度为 O(log(N))）或任何其他的时间复杂度低于 O(N) 的数据结构。
```
Comment (yin1999) [inline files/zh-cn/web/javascript/reference/global_objects/set/index.md:14]:
```suggestion
规范要求集合的实现是“对合集中的元素的平均访问时间与集合中元素的数量呈次线性关系”。因此，它可以在内部表示为哈希表（查找的时间复杂度为 O(1)）、搜索树（查找的时间复杂度为 O(log(N))）或任何其他的时间复杂度低于 O(N) 的数据结构。
```

File: files/zh-cn/web/javascript/reference/global_objects/set/index.md [hunk 3]
```diff
@@ -3,131 +3,181 @@ title: Set
...
 
 ### 值的相等
 
-因为 Set 中的值总是唯一的，所以需要判断两个值是否相等。在 ECMAScript 规范的早期版本中，这不是基于和===操作符中使用的算法相同的算法。具体来说，对于 Set，+0（+0 严格相等于 -0）和 -0 是不同的值。然而，在 ECMAScript 2015 规范中这点已被更改。有关详细信息，请参阅[浏览器兼容性](#浏览器兼容性)表中的“_Key equality for -0 and 0_”。
+值的相等是基于[零值相等](/zh-CN/docs/Web/JavaScript/Equality_comparisons_and_sameness#零值相等)算法的。（曾经使用[同值算法](/zh-CN/docs/Web/JavaScript/Equality_comparisons_and_sameness#使用_object.is_进行同值相等比较)，会将 `0` 和 `-0` 视为不同值。参见[浏览器兼容性](#浏览器兼容性)。）这意味着 {{jsxref("NaN")}} 和 {{jsxref("undefined")}} 会被视为是相同的值（即使 `NaN !== NaN`），而所有其他的值则基于 `===` 运算符的语义进行相等比较。
```
Comment (yin1999) [inline files/zh-cn/web/javascript/reference/global_objects/set/index.md:18]:
```suggestion
值的相等是基于[零值相等](/zh-CN/docs/Web/JavaScript/Equality_comparisons_and_sameness#零值相等)算法的。（曾使用会将 `0` 和 `-0` 视为不同值的[同值算法](/zh-CN/docs/Web/JavaScript/Equality_comparisons_and_sameness#使用_object.is_进行同值相等比较)。参见[浏览器兼容性](#浏览器兼容性)。）这意味着 {{jsxref("NaN")}} 和 {{jsxref("undefined")}} 会被视为是相同的值（即使 `NaN !== NaN`），而所有其他的值则基于 `===` 运算符的语义进行相等比较。
```

File: files/zh-cn/web/javascript/reference/global_objects/set/index.md [hunk 4]
```diff
@@ -3,131 +3,181 @@ title: Set
...
+
+### 性能
+
+[`has`](/zh-CN/docs/Web/JavaScript/Reference/Global_Objects/Set/has) 方法检查某个值是否在集合中，使用的方法的平均时间比逐个测试先前添加到集合中的元素更快。特别是，当一个数组的长度（`length`）等于集合的大小（`size`）时，这比 [`Array.prototype.includes`](/zh-CN/docs/Web/JavaScript/Reference/Global_Objects/Array/includes) 方法的平均速度更快。
```
Comment (yin1999) [inline files/zh-cn/web/javascript/reference/global_objects/set/index.md:22]:
```suggestion
[`has`](/zh-CN/docs/Web/JavaScript/Reference/Global_Objects/Set/has) 方法检查某个值是否在集合中，其采用的方式的平均时间比逐个测试先前添加到集合中的元素更快。特别是当一个数组的长度（`length`）等于集合的大小（`size`）时，这比 [`Array.prototype.includes`](/zh-CN/docs/Web/JavaScript/Reference/Global_Objects/Array/includes) 方法的平均速度更快。
```

File: files/zh-cn/web/javascript/reference/global_objects/set/index.md [hunk 5]
```diff
@@ -3,131 +3,181 @@ title: Set
...
+
+### 类 Set 浏览器 API
+
+浏览器**类 `Set` 对象**（或“类集合对象”）是行为与 `Set` 类似的 [Web API](/zh-CN/docs/Web/API) 接口。
+
+就像 `Set` 一样，元素可以按照它们被添加到对象的顺序进行迭代。类 `Set` 对象和 `Set` 也有相同名称和行为的属性和方法。但是，与 `Set` 不同的是，前者只允许每个条目特定的预定义类型。
```
Comment (yin1999) [inline files/zh-cn/web/javascript/reference/global_objects/set/index.md:28]:
```suggestion
就像 `Set` 一样，元素可以按照它们被添加到对象的顺序进行迭代。类 `Set` 对象也有与 `Set` 相同名称和行为的属性及方法。但是，与 `Set` 不同的是，前者只允许每个条目特定的预定义类型。
```

File: files/zh-cn/web/javascript/reference/global_objects/set/index.md [hunk 6]
```diff
@@ -3,131 +3,181 @@ title: Set
...
+
+```webidl
+interface GPUSupportedFeatures {
+  readonly setlike<DOMString>;
+};
+```
+
+类 `Set` 对象是只读或可读写的（参见上面的 IDL 中的 `readonly` 关键字）。
+
+- 只读的类 `Set` 对象具有 [`size`](#set.prototype.size) 属性和以下方法：[`entries()`](#set.prototype.entries)、[`forEach()`](#set.prototype.foreach)、[`has()`](#set.prototype.has)、[`keys()`](#set.prototype.keys)、[`values()`](#set.prototype.values) 以及 [`@@iterator`](#set.prototypeiterator)。
```
Comment (yin1999) [inline files/zh-cn/web/javascript/reference/global_objects/set/index.md:40]:
```suggestion
- 只读的类 `Set` 对象具有 [`size`](#set.prototype.size) 属性及以下方法：[`entries()`](#set.prototype.entries)、[`forEach()`](#set.prototype.foreach)、[`has()`](#set.prototype.has)、[`keys()`](#set.prototype.keys)、[`values()`](#set.prototype.values) 和 [`@@iterator`](#set.prototypeiterator)。
```

File: files/zh-cn/web/javascript/reference/global_objects/set/index.md [hunk 7]
```diff
@@ -3,131 +3,181 @@ title: Set
...
-  - : 返回一个新的迭代器对象，该对象包含 `Set` 对象中的按插入顺序排列的所有元素的值的 `[value, value]` 数组。为了使这个方法和 `Map` 对象保持相似，每个值的键和值相等。
-- {{jsxref("Set.forEach", "Set.prototype.forEach(<em>callbackFn</em>[, <em>thisArg</em>])")}}
-  - : 按照插入顺序，为 Set 对象中的每一个值调用一次 callBackFn。如果提供了`thisArg`参数，回调中的 `this` 会是这个参数。
-- {{jsxref("Set.has", "Set.prototype.has(<em>value</em>)")}}
-  - : 返回一个布尔值，表示该值在 `Set` 中存在与否。
+  - : 返回一个新的迭代器对象，该对象包含 `Set` 对象中的代表每个元素的 **`[value, value]` 数组**。这与 {{jsxref("Map")}} 对象类似，因此 `Set` 的每个条目的 _key_ 和 _value_ 都相同。
+- {{jsxref("Set.prototype.forEach()")}}
+  - : 按照值插入的顺序为 `Set` 对象中的每个值调用一次 `callbackFn`。如果提供了 `thisArg` 参数，它将被用作每次调用 `callbackFn` 时的 `this` 值。
+- {{jsxref("Set.prototype.has()")}}
+  - : 返回一个布尔值，表示该值在 `Set` 对象中是否存在。
```
Comment (yin1999) [inline files/zh-cn/web/javascript/reference/global_objects/set/index.md:90]:
```suggestion
  - : 返回一个布尔值，表示给定值在 `Set` 对象中是否存在。
```

File: files/zh-cn/web/javascript/reference/global_objects/set/index.md [hunk 8]
```diff
@@ -3,131 +3,181 @@ title: Set
...
-  - : 返回一个新的迭代器对象，该对象包含 `Set` 对象中的按插入顺序排列的所有元素的值。
-- {{jsxref("Set.prototype.@@iterator()", "Set.prototype[@@iterator]()")}}
-  - : 返回一个新的迭代器对象，该对象包含 `Set` 对象中的按插入顺序排列的所有元素的值。
+  - : 返回一个新的迭代器对象，该对象按插入顺序生成 `Set` 对象中每个元素的**值**。
+- [`Set.prototype[@@iterator]()`](/en-US/docs/Web/JavaScript/Reference/Global_Objects/Set/@@iterator)
+  - : 返回一个新的迭代器对象，该对象按插入顺序生成 `Set` 对象中每个元素的**值**。
 
 ## 示例
 
 ### 使用 `Set` 对象
```
Comment (yin1999) [inline files/zh-cn/web/javascript/reference/global_objects/set/index.md:100]:
```suggestion
### 使用 Set 对象
```

File: files/zh-cn/web/javascript/reference/global_objects/set/index.md [hunk 9]
```diff
@@ -3,131 +3,181 @@ title: Set
...
+  - : 返回一个布尔值，表示给定值在 `Set` 对象中是否存在。
 - {{jsxref("Set.prototype.keys()")}}
-  - : 与 **`values()`** 方法相同，返回一个新的迭代器对象，该对象包含 `Set` 对象中的按插入顺序排列的所有元素的值。
+  - : {{jsxref("Set.prototype.values()")}} 的别名。
 - {{jsxref("Set.prototype.values()")}}
-  - : 返回一个新的迭代器对象，该对象包含 `Set` 对象中的按插入顺序排列的所有元素的值。
-- {{jsxref("Set.prototype.@@iterator()", "Set.prototype[@@iterator]()")}}
-  - : 返回一个新的迭代器对象，该对象包含 `Set` 对象中的按插入顺序排列的所有元素的值。
+  - : 返回一个新的迭代器对象，该对象按插入顺序生成 `Set` 对象中每个元素的**值**。
+- [`Set.prototype[@@iterator]()`](/en-US/docs/Web/JavaScript/Reference/Global_Objects/Set/@@iterator)
```
Comment (jasonren0403) [inline files/zh-cn/web/javascript/reference/global_objects/set/index.md:95]:
```suggestion
- [`Set.prototype[@@iterator]()`](/zh-CN/docs/Web/JavaScript/Reference/Global_Objects/Set/@@iterator)
```

**Review by jasonren0403 — APPROVED** [review]
Thanks 👍 Just one minor fix

**Pull request merged**